src/lib.rs

```diff
@@ -3,6 +3,16 @@
 //!
 //! Welcome to typesense, the rust library for the Typesense API.
 
+pub mod collection;
 mod error;
+pub mod field;
 
 pub use error::{Result, TypesenseError};
+
+#[cfg(feature = "typesense_derive")]
+#[allow(unused_imports)]
+#[macro_use]
+extern crate typesense_derive;
```

Comment:
Is `extern crate` required, you could import the macros with only the `use` statement right?